source/common/upstream/upstream_impl.h

```diff
@@ -1059,24 +1063,41 @@ class ClusterInfoImpl : public ClusterInfo,
   static std::pair<absl::optional<double>, absl::optional<uint32_t>>
   getRetryBudgetParams(const envoy::config::cluster::v3::CircuitBreakers::Thresholds& thresholds);
 
+  static envoy::config::core::v3::TypedExtensionConfig getRetryAdmissionControlConfig(
+      const absl::optional<envoy::config::cluster::v3::CircuitBreakers::Thresholds>& thresholds);
+
 private:
   std::shared_ptr<UpstreamNetworkFilterConfigProviderManager>
   createSingletonUpstreamNetworkFilterConfigProviderManager(
       Server::Configuration::ServerFactoryContext& context);
 
-  struct ResourceManagers {
-    ResourceManagers(const envoy::config::cluster::v3::Cluster& config, Runtime::Loader& runtime,
-                     const std::string& cluster_name, Stats::Scope& stats_scope,
-                     const ClusterCircuitBreakersStatNames& circuit_breakers_stat_names);
-    ResourceManagerImplPtr load(const envoy::config::cluster::v3::Cluster& config,
-                                Runtime::Loader& runtime, const std::string& cluster_name,
-                                Stats::Scope& stats_scope,
-                                const envoy::config::core::v3::RoutingPriority& priority);
-
-    using Managers = std::array<ResourceManagerImplPtr, NumResourcePriorities>;
-
-    Managers managers_;
-    const ClusterCircuitBreakersStatNames& circuit_breakers_stat_names_;
+  struct CircuitBreakers {
```

Comment:
This struct changed a bit to accommodate managing both the resource managers and the admission controls. This is because the stats struct, priority mapping, and runtime key prefix calculations are shared between both.